qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py

```diff
@@ -295,7 +295,15 @@ def update_clim_validator(self):
     def _autoscale_clim(self):
         """Update stored colorbar limits
         The new limits are found from the colobar data """
-        data = self.colorbar.mappable.get_array()
+        data = self.colorbar.mappable.get_array_clipped_to_bounds()
+
+        # If any dimension is zero then we have no data in the display area
+        if any(map(lambda dim: dim == 0, data.shape)):
```

Comment:
This is a really good addition, I believe it works for MD Histo workspaces (I think it would work for MDEvent anyway becasue BinMD get recalled even for limits outside the extent of the data but it's hard to confirm for MDHisto workspaces as autoscale previously didn't work at all!).
However, I've found that in a Workspace2D (e.g. `MAR11060.raw` which has a spike in the last bin of every spectrum which makes this bug easier to reproduce) if you zoom outside the extent of the data the color limits don't seem to update to (cmax, cmin) = (0,0) it gets frozen at whatever the limits were when there was still data on the image. Or maybe there is a bug elsewhere?

Reply:
I've moved the check to the masked data rather than the original data which seems to fix this.